app/com/yetu/oauth2provider/oauth2/handlers/AuthorizationHandler.scala

```diff
@@ -86,23 +82,26 @@ class AuthorizationHandler(authAccessService: IAuthCodeAccessTokenService,
   }
 
   def findUser(username: String, password: String): Future[Option[YetuUser]] = {
-    logger.warn("Find user... ")
+    personService.findUser(username).map {
+      case Some(user) =>
 
-    val loggedIn = for {
-      user <- personService.findYetuUser(username)
-      pinfo <- user.passwordInfo
-      hasher <- passwordHashers.get(pinfo.hasher) if hasher.matches(pinfo, password)
-    } yield user
+        val itMatches = user.passwordInfo
+          .flatMap(pInfo => passwordHashers.get(pInfo.hasher)
+            .map(_.matches(pInfo, password)))
+          .getOrElse(false)
```

Comment:
instead of `getOrElse` and the `if`below, just use a `map(user)` at the end of this line